avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1

```diff
@@ -0,0 +1,75 @@
+﻿<#
+.SYNOPSIS
+Checks new PRs and adds labels, depending if module owner can approve, or core team is needed
+
+.DESCRIPTION
+For the given PR, the "needs core team" label will be added, if the module is orphaned or the sole module owner is creating the PR
+
+.PARAMETER Repo
+Mandatory. The name of the respository to scan. Needs to have the structure "<owner>/<repositioryName>", like 'Azure/bicep-registry-modules/'
+
+.PARAMETER RepoRoot
+Optional. Path to the root of the repository.
+
+.PARAMETER PrUrl
+Mandatory. The URL of the GitHub pull request, like 'https://github.com/Azure/bicep-registry-modules/pull/2540'
+
+.EXAMPLE
+Set-AvmGitHubPrLabels -Repo 'Azure/bicep-registry-modules' -PrUrl 'https://github.com/Azure/bicep-registry-modules/pull/2540'
+
+.NOTES
+Will be triggered by the workflow avm.platform.set-avm-github-pr-labels.yml
+#>
+function Set-AvmGitHubPrLabels {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $Repo,
+
+        [Parameter(Mandatory = $true)]
+        [string] $PrUrl,
+
+        [Parameter(Mandatory = $false)]
+        [string] $RepoRoot = (Get-Item -Path $PSScriptRoot).parent.parent.parent.parent.FullName
+    )
+
+    # Loading helper functions
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-GithubPrRequestedReviewerTeamNames.ps1')
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-GithubTeamMembersLogin.ps1')
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-AvmCsvData.ps1')
+
+    $pr = gh pr view $PrUrl.Replace('api.', '').Replace('repos/', '') --json 'author,title,url,body,comments' --repo $Repo | ConvertFrom-Json -Depth 100
+    $allTeamNames = [array](Get-GithubPrRequestedReviewerTeamNames -PrUrl $PrUrl.Replace('api.', '').Replace('repos/', ''))
+    $teamNames = [array]($allTeamNames | Where-Object { $_ -ne 'bicep-admins' -and $_ -ne 'avm-core-team-technical-bicep' -and $_ -ne 'avm-module-reviewers-bicep' })
+
+    # check for orphanded module
+    if ($teamMembers.Count -eq 1) {
+        $moduleName = $teamMembers[0]
+        $moduleIndex = $moduleName.StartsWith('avm-res') ? 'Bicep-Resource' : 'Bicep-Pattern'
+        # get CSV data
+        $modules = Get-AvmCsvData -ModuleIndex $moduleIndex
```

Comment:
If I get it correctly we are retrieving the whole list of modules from the csv and doing a replace and comparison operation for each. I think the cost of this should be optimized. I'd suggest to instead retrieve the module name from the team name and just do one operation in checking if that team status is orphaned

Reply:
As we do not use Voyager 2 and have extreme limitations, I think we should not care. We are talking about operations that take milliseconds. Joke aside, what do you have in mind?

Reply:
Fair, in case we can improve later